src/fs/metadata_log.hh

```diff
@@ -65,6 +66,11 @@ class metadata_log {
     inode_t _root_dir;
     shard_inode_allocator _inode_allocator;
 
+    // Estimations of metadata log size used in compaction
+    size_t _ondisk_log_size = 0;
+    size_t _rewrite_log_size = 0;
+    std::unordered_map<cluster_id_t, cluster_info> _clusters;
```

Comment:
Maybe instead of keeping record if cluster is writable or not in the cluster itself, splitting this into two:
- _writable_clusters
- _read_only_clusters
?
Ofc, that is just an idea not a "requested change".